packages/types/src/storages.ts

```diff
@@ -223,6 +223,27 @@ export interface ListOptions {
     limit?: number;
 }
 
+export interface ListAndLockOptions extends ListOptions {
+    lockSecs: number;
+}
+
+export interface ListAndLockHeadResult extends QueueHead {
+    lockSecs: number;
+}
+
+export interface ProlongRequestLockOptions {
+    lockSecs: number;
+    forefront?: boolean;
+}
+
+export interface ProlongRequestLockResult {
+    lockExpiresAt: Date;
```

Comment:
Discussion: what does the api client return? Does it also return a date object, or just the date string?